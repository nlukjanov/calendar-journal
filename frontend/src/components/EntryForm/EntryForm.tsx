import React from 'react';

import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';

type EntryForm = {
  formData: any;
  handleSubmit: any;
  handleChange: any;
};

const EntryForm = ({ formData, handleSubmit, handleChange }: EntryForm) => {
  console.log('formData', formData);
  return (
    <Card className='mx-auto mt-5 border-light' style={{ maxWidth: '25rem' }}>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId='title'>
          <Form.Label>Title</Form.Label>
          <Form.Control
            name='title'
            type='title'
            placeholder='Enter title'
            value={formData.title}
            onChange={handleChange}
          />
        </Form.Group>
        <Form.Group controlId='entryText'>
          <Form.Label>Entry Text</Form.Label>
          <Form.Control
            type='textarea'
            name='entryText'
            placeholder='Entry Text'
            value={formData.entryText}
            onChange={handleChange}
          />
        </Form.Group>
        <Form.Group controlId='date'>
          <Form.Label>Date</Form.Label>
          <Form.Control
            type='datetime-local'
            name='date'
            placeholder='Date'
            value={formData.date}
            onChange={handleChange}
          />
        </Form.Group>
        <Button variant='primary' type='submit'>
          Submit
        </Button>
      </Form>
    </Card>
  );
};

export default EntryForm;
