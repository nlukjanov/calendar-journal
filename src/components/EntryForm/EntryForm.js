import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import axios from 'axios';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import { getToken } from '../../lib/authHelper';

const EntryForm = ({ date }) => {
  const history = useHistory();
  const passedDate = date ? new Date(date).toISOString().substr(0, 16) : '';
  const [formData, setFormData] = useState({
    title: '',
    entryText: '',
    date: passedDate,
  });
  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await axios.post('http://localhost:4000/api/journal', formData, {
        headers: { Authorization: `Bearer ${getToken('token')}` },
      });
      history.push('/myjournal');
      // add notification with res
    } catch (error) {
      console.log(error);
    }
  };

  const handleChange = ({ target: { name, value } }) => {
    const formNewData = { ...formData, [name]: value };
    setFormData(formNewData);
  };

  useEffect(() => {
    console.log(formData.date);
  }, [formData.date]);

  return (
    <Card className='mx-auto mt-5 border-light' style={{ maxWidth: '25rem' }}>
      <Form onSubmit={handleSubmit}>
        <Form.Group controlId='title'>
          <Form.Label>Title</Form.Label>
          <Form.Control
            name='title'
            type='title'
            placeholder='Enter title'
            onChange={handleChange}
          />
        </Form.Group>
        <Form.Group controlId='entryText'>
          <Form.Label>Entry Text</Form.Label>
          <Form.Control
            type='textarea'
            name='entryText'
            placeholder='Entry Text'
            onChange={handleChange}
          />
        </Form.Group>
        <Form.Group controlId='date'>
          <Form.Label>Date</Form.Label>
          <Form.Control
            type='datetime-local'
            name='date'
            placeholder='Date'
            onChange={handleChange}
          />
        </Form.Group>
        <Button variant='primary' type='submit'>
          Create Entry
        </Button>
      </Form>
    </Card>
  );
};

export default EntryForm;
