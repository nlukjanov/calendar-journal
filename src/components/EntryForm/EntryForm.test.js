import React from 'react';
// eslint-disable-next-line node/no-unpublished-import
import { render } from '@testing-library/react';
import EntryForm from './EntryForm';

test('renders new journal entry', async () => {
  const { container } = render(<EntryForm />);
  expect(container).toMatchSnapshot();
});
