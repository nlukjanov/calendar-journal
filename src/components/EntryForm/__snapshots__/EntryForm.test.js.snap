// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders new journal entry 1`] = `
<div>
  <div
    class="mx-auto mt-5 border-light card"
    style="max-width: 25rem;"
  >
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="title"
        >
          Title
        </label>
        <input
          class="form-control"
          id="title"
          name="title"
          placeholder="Enter title"
          type="title"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="entryText"
        >
          Entry Text
        </label>
        <input
          class="form-control"
          id="entryText"
          name="entryText"
          placeholder="Entry Text"
          type="textarea"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="date"
        >
          Date
        </label>
        <input
          class="form-control"
          id="date"
          name="date"
          placeholder="Date"
          type="datetime-local"
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Create Entry
      </button>
    </form>
  </div>
</div>
`;
